database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java

```diff
@@ -165,7 +165,7 @@ static String removeConditionsClause(String importQueryString) {
     return query;
   }
 
-  private Schema loadSchemaFromDB(Class<? extends Driver> driverClass)
+  public Schema loadSchemaFromDB(Class<? extends Driver> driverClass)
```

Comment:
This method doesn't seem a good candidate to be public. The one above seems to be more appropriate for public. 

Can the method be simply loadSchemaFromDB(connection) ? What's is query string used for?